data/darktableconfig.xml.in

```diff
@@ -1341,6 +1334,39 @@
     <shortdescription>high quality processing from size</shortdescription>
     <longdescription>if the thumbnail size is greater than this value, it will be processed using the full quality rendering path (better but slower).\nif you want all thumbnails and pre-rendered images in best quality you should choose the *always* option.\n(more comments in the manual)</longdescription>
   </dtconfig>
+  <dtconfig prefs="lighttable" section="thumbs">
+    <name>cache_disk_backend</name>
+    <type>bool</type>
+    <default>true</default>
+    <shortdescription>enable disk backend for thumbnail cache</shortdescription>
+    <longdescription>if enabled, write thumbnails to disk (.cache/darktable/) when evicted from the memory cache.\nnote that this can take a lot of memory (several gigabytes for 20k images) and will never delete cached thumbnails again.\nit's safe though to delete these manually, if you want. light table performance will be increased greatly when browsing a lot.\nto generate all thumbnails of your entire collection offline, run 'darktable-generate-cache'.</longdescription>
+  </dtconfig>
+  <dtconfig prefs="lighttable" section="thumbs">
+    <name>cache_disk_backend_full</name>
+    <type>bool</type>
+    <default>false</default>
+    <shortdescription>enable disk backend for full preview cache</shortdescription>
+    <longdescription>if enabled, write full preview to disk (.cache/darktable/) when evicted from the memory cache.\nnote that this can take a lot of memory (several gigabytes for 20k images) and will never delete cached thumbnails again.\nit's safe though to delete these manually, if you want.\nlight table performance will be increased greatly when zooming image in full preview mode.</longdescription>
+  </dtconfig>
+  <dtconfig prefs="lighttable" section="thumbs">
+    <name>backthumbs_mipsize</name>
+    <type>
+      <enum>
+        <option>never</option>
+        <option>small</option>
+        <option>VGA</option>
+        <option>720p</option>
+        <option>1080p</option>
+        <option>WQXGA</option>
+        <option>4K</option>
+        <option>5K</option>
+        <option>reinitialize database</option>
+      </enum>
+    </type>
+    <default>never</default>
+    <shortdescription>generate thumbs in background</shortdescription>
+    <longdescription>if 'cache_disk_backend' is enabled thumbnails/mipmaps up to the selected size are generated while user is inactive in lighttable. 'reinitialize database' restarts the database</longdescription>
```

Comment:
For `reinitialize database` we will want a better explanation. See my question about why/when this is needed. We will then try to give a better wording here.

Reply:
Given the discussion my proposal is to have another pref (a boolean witch checkbox) called `reset database thumbs status` (or a better proposal). Do the reset when returning from the pref and unset back this pref as we want to reset once. And so we can remove `<option>reinitialize database</option>` above which doesn't fit well into the list.

Reply:
right. i had initially hoped to get along with one preference setting ...